### DISCUSSION TITLE
zfswin-2.2.0rc6 testing

### DISCUSSION THREAD

**Discussion by lundman** [openzfsonwindows/openzfs#321] [Testing]
1:
Creating:
BOOM
BOOM/lower
BOOM/lowerer

Had inconsistent `zfs mount` output.

2:
rclone sync . E:/lower/
does not work at all. Could it be rclone does not handle reparse-points? Test with vhd.

3:
snapshot mount showed directory, but no contents, bug?

4:
zpool export fails with "busy". Because snapshot mount went wrong? bug?

**Comment by EchterAgo** [comment]
I also have found another issue with the export/unmount.

So I import an existing dataset that has my backups on it, then I let rclone sync. It just checks the metadata of ~95k files and does not write anything new because things are already up to date. Then I try to export the pool and it will just hang forever.

What is interesting here is that all the `VREG` nodes get reclaimed very fast but then on the `filesanddirs` pass there are only directory nodes (`VDIR`) left and they can not be reclaimed because of `v_usecount > 0`. I let this run for quite a while and just nothing changes. I also checked if there are still any programs that have handles to the path open, but there are none.

These notes all have `v_type == VDIR`, `v_usecount > 0`, `v_iocount == 0`, `SectionObjectPointers == NULL`, `v_data == NULL`, except for the `VNODE_MARKROOT` nodes.

Is there a way to diagnose what is keeping their `v_usecount > 0`?

**Reply by EchterAgo** [reply]
Nevermind, I think what I'm seeing here is `vflush` just taking a long time to process 95k `vnode`s. I was able to optimize this a lot by reducing the locking calls in the loop, I'll try to make a PR for that.

**Reply by EchterAgo** [reply]
I managed to get the reclaim time for the 95k `vnode`s down from minutes to a second.

**Reply by EchterAgo** [reply]
See #328
